wsd/DocumentBroker.cpp

```diff
@@ -2772,6 +2776,12 @@ bool DocumentBroker::forwardToChild(const std::string& viewId, const std::string
     const auto it = _sessions.find(viewId);
     if (it != _sessions.end())
     {
+        if (!it->second->isReadOnly() && it->second->isViewLoaded() &&
```

Comment:
Not a fan of this as a location for this work. I would expect to hook into DocumentBroker::updateLastActivityTime - and get the client-session to pass its own const &view-id into that. I would expect to compare with what we already have, so as to avoid duplicating the std::string on the heap very frequently.

With that fix - the approach looks ok to me.

Thanks !

Reply:
> Not a fan of this as a location for this work. I would expect to hook into DocumentBroker::updateLastActivityTime - and get the client-session to pass its own const &view-id into that.

Unfortunately, the `AdminModel` also calls `DocumentBroker::updateLastActivityTime` (but it doesn't have a session to pass).

I can add another similar helper for this, so we can do it as you describe, but without changing `updateLastActivityTime`.

> I would expect to compare with what we already have, so as to avoid duplicating the std::string on the heap very frequently.

Sure.